### PR TITLE
gitquery: implement session

### PR DESCRIPTION
**Pull request by erizocosmico**
Depends on https://github.com/src-d/go-mysql-server/issues/36 and https://github.com/src-d/go-mysql-server/pull/87 (that's why CI fails)

### CODE REVIEW

File: internal/function/is_remote.go [hunk 1]
```diff
@@ -18,11 +18,9 @@ func NewIsRemote(e sql.Expression) sql.Expression {
 	return &IsRemote{expression.UnaryExpression{Child: e}}
 }
 
-var _ sql.Expression = (*IsRemote)(nil)
```
Comment (mcarmonaa) [inline internal/function/is_remote.go]: Why now isn't needed to check that it implements the `sql.Expression` interface?

File: internal/function/is_tag.go [hunk 2]
```diff
@@ -18,11 +18,9 @@ func NewIsTag(e sql.Expression) sql.Expression {
 	return &IsTag{expression.UnaryExpression{Child: e}}
 }
 
-var _ sql.Expression = (*IsTag)(nil)
```
Comment (mcarmonaa) [inline internal/function/is_tag.go]: Why now isn't needed to check that it implements the `sql.Expression` interface?

File: internal/function/is_remote.go (hunk 1 above)
Comment (erizocosmico) [reply internal/function/is_remote.go]: `TransformUp` already makes this check

**Comment by ajnavarro** [comment]
should we merge this?

**Comment by erizocosmico** [comment]
@ajnavarro once I rebase, yes

**Pull request merged**